src/bulkOperationCommand.ts

```diff
@@ -133,10 +153,20 @@ export abstract class BulkOperationCommand extends BulkBaseCommand {
         }
         this.displayBulkV2Result(jobInfo);
         const result = { jobInfo } as BulkResultV2;
-        if (!isBulkV2RequestDone(jobInfo) || !this.jsonEnabled()) {
+        if (!isBulkV2RequestDone(jobInfo)) {
+          return result;
+        }
+        let records = null;
+        if (verbose) {
```

Comment:
could this be moved to `displayBulkV2Result`?

Reply:
It could be moved, yes. But then we'd need to change the signature of the `BulkBaseCommand` function to something like this: `async displayBulkV2Result(jobInfo: JobInfoV2, verbose: boolean = false): Promise<void>` because
* We might need to fetch results in that function (`async`)
* Function is also used in `ResumeBulkCommand.resume`, which does not have a `verbose` flag (so adding a default `boolean` parameter for `verbose`)

Also we'd then need to move `printBulkErrors` to the `BulkBaseCommand` class. Not sure if that makes sense since we're defining the parameter in the descendant `BulkOperationCommand`?

And finally, we might pull the results twice (`await this.job.getAllResults()`) if the `verbose` flag is true. One time inside of the `BulkBaseCommand.displayBulkV2Result` and one time again in this function here (`BulkOperationCommand.runBulkOperation`).

So maybe something like that would be better:

```js
        this.displayBulkV2Result(jobInfo);
        const result = { jobInfo } as BulkResultV2;
        if (!isBulkV2RequestDone(jobInfo)) {
          return result;
        }
        if (!this.jsonEnabled()) {
          if (verbose) {
            const records = await this.job.getAllResults();
            if (records?.failedResults?.length > 0) {
              this.printBulkErrors(records.failedResults);
            }
          }
          return result;
        }
        result.records = transformResults(await this.job.getAllResults());
        return result;
```